Add kokkos+libceed to all petsc builds, update container mpi, update containers, fix conda zsh on linux

loganharbour opened the pull request:
- Adds `--download-kokkos` and `--download-kokkos-kernels` to all PETSc builds (including CPU)
- Updates openmpi in all containers to openmpi 5.0.8
- Updates mpich in all containers to mpich 4.3.1
- Updates base container (system packages)
- Adds `--download-libceed` to all PETSc builds and uses it in MFEM (closes https://github.com/idaholab/moose/issues/31089)
- Fixes conda zsh prompt on linux (closes https://github.com/idaholab/moose/issues/31041)

loganharbour committed:
Add kokkos to CPU petsc build

refs #0

loganharbour committed:
Bump mpi, libmesh-vtk, petsc, libmesh, moose-dev

refs #0

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3063595/), step Versioner verify on 3666f52 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 7 changed, 0 added, 0 removed

| package   | status   | file                                    |
|-----------|----------|-----------------------------------------|
| petsc     | CHANGE   | apptainer/petsc.def                     |
| petsc     | CHANGE   | conda/petsc/meta.yaml                   |
| petsc     | CHANGE   | scripts/configure_petsc.sh              |
| libmesh   | CHANGE   | conda/libmesh/conda_build_config.yaml   |
| libmesh   | CHANGE   | conda/libmesh/meta.yaml                 |
| moose-dev | CHANGE   | conda/moose-dev/conda_build_config.yaml |
| moose-dev | CHANGE   | conda/moose-dev/meta.yaml               |

### Versioner versions

Found 9 packages, 3 changed, 0 failed

| package   | status   | hash    | to hash   | version                       | to version                    |
|-----------|----------|---------|-----------|-------------------------------|-------------------------------|
| petsc     | CHANGE   | 9531950 | c452ee6   | 3.23.0.6.gd9d7fd11dca build 2 | 3.23.0.6.gd9d7fd11dca build 3 |
| libmesh   | CHANGE   | 985bf51 | 795e5fe   | 2025.06.25 build 0            | 2025.06.25 build 1            |
| moose-dev | CHANGE   | b5c6af5 | 74de705   | 2025.06.26                    | 2025.07.22                    |

Heinrich-BR committed:
Add libceed download to petsc (closes #31089)

loganharbour committed:
Update base containers

- Updates mpich to 4.3.1 for all mpich builds
- Adds mpich to cuda builds
- Updates openmpi to 5.0.8 for all builds

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3064746/), step Versioner verify on 4a65741 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 12 changed, 0 added, 0 removed

| package     | status   | file                                      |
|-------------|----------|-------------------------------------------|
| mpi         | CHANGE   | apptainer/mpi.def                         |
| mpi         | CHANGE   | conda/mpi/meta.yaml                       |
| libmesh-vtk | CHANGE   | conda/libmesh-vtk/conda_build_config.yaml |
| libmesh-vtk | CHANGE   | conda/libmesh-vtk/meta.yaml               |
| petsc       | CHANGE   | apptainer/petsc.def                       |
| petsc       | CHANGE   | conda/petsc/conda_build_config.yaml       |
| petsc       | CHANGE   | conda/petsc/meta.yaml                     |
| petsc       | CHANGE   | scripts/configure_petsc.sh                |
| libmesh     | CHANGE   | conda/libmesh/conda_build_config.yaml     |
| libmesh     | CHANGE   | conda/libmesh/meta.yaml                   |
| moose-dev   | CHANGE   | conda/moose-dev/conda_build_config.yaml   |
| moose-dev   | CHANGE   | conda/moose-dev/meta.yaml                 |

### Versioner versions

Found 9 packages, 5 changed, 0 failed

| package     | status   | hash    | to hash   | version                       | to version                    |
|-------------|----------|---------|-----------|-------------------------------|-------------------------------|
| mpi         | CHANGE   | fcdc117 | 94cfd77   | 2025.06.13                    | 2025.07.23                    |
| libmesh-vtk | CHANGE   | 16265eb | df9bbbd   | 9.4.2 build 3                 | 9.4.2 build 4                 |
| petsc       | CHANGE   | 9531950 | 36d68f6   | 3.23.0.6.gd9d7fd11dca build 2 | 3.23.0.6.gd9d7fd11dca build 3 |
| libmesh     | CHANGE   | 985bf51 | 57b7481   | 2025.06.25 build 0            | 2025.06.25 build 1            |
| moose-dev   | CHANGE   | b5c6af5 | 071c3fd   | 2025.06.26                    | 2025.07.22                    |

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3065391/), step Versioner verify on cd02e39 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 12 changed, 0 added, 0 removed

| package     | status   | file                                      |
|-------------|----------|-------------------------------------------|
| mpi         | CHANGE   | apptainer/mpi.def                         |
| mpi         | CHANGE   | conda/mpi/meta.yaml                       |
| libmesh-vtk | CHANGE   | conda/libmesh-vtk/conda_build_config.yaml |
| libmesh-vtk | CHANGE   | conda/libmesh-vtk/meta.yaml               |
| petsc       | CHANGE   | apptainer/petsc.def                       |
| petsc       | CHANGE   | conda/petsc/conda_build_config.yaml       |
| petsc       | CHANGE   | conda/petsc/meta.yaml                     |
| petsc       | CHANGE   | scripts/configure_petsc.sh                |
| libmesh     | CHANGE   | conda/libmesh/conda_build_config.yaml     |
| libmesh     | CHANGE   | conda/libmesh/meta.yaml                   |
| moose-dev   | CHANGE   | conda/moose-dev/conda_build_config.yaml   |
| moose-dev   | CHANGE   | conda/moose-dev/meta.yaml                 |

### Versioner versions

Found 9 packages, 5 changed, 0 failed

| package     | status   | hash    | to hash   | version                       | to version                    |
|-------------|----------|---------|-----------|-------------------------------|-------------------------------|
| mpi         | CHANGE   | fcdc117 | 334c18b   | 2025.06.13                    | 2025.07.23                    |
| libmesh-vtk | CHANGE   | 16265eb | 922630b   | 9.4.2 build 3                 | 9.4.2 build 4                 |
| petsc       | CHANGE   | 9531950 | 7a13ad8   | 3.23.0.6.gd9d7fd11dca build 2 | 3.23.0.6.gd9d7fd11dca build 3 |
| libmesh     | CHANGE   | 985bf51 | fee6d95   | 2025.06.25 build 0            | 2025.06.25 build 1            |
| moose-dev   | CHANGE   | b5c6af5 | 949a2e7   | 2025.06.26                    | 2025.07.22                    |

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3066011/), step Versioner verify on 9482074 wanted to post the following:

### Versioner templates

Found 14 templates, 0 failed


### Versioner influential files

Found 58 influential files, 12 changed, 0 added, 0 removed

| package     | status   | file                                      |
|-------------|----------|-------------------------------------------|
| mpi         | CHANGE   | apptainer/mpi.def                         |
| mpi         | CHANGE   | conda/mpi/meta.yaml                       |
| libmesh-vtk | CHANGE   | conda/libmesh-vtk/conda_build_config.yaml |
| libmesh-vtk | CHANGE   | conda/libmesh-vtk/meta.yaml               |
| petsc       | CHANGE   | apptainer/petsc.def                       |
| petsc       | CHANGE   | conda/petsc/conda_build_config.yaml       |
| petsc       | CHANGE   | conda/petsc/meta.yaml                     |
| petsc       | CHANGE   | scripts/configure_petsc.sh                |
| libmesh     | CHANGE   | conda/libmesh/conda_build_config.yaml     |
| libmesh     | CHANGE   | conda/libmesh/meta.yaml                   |
| moose-dev   | CHANGE   | conda/moose-dev/conda_build_config.yaml   |
| moose-dev   | CHANGE   | conda/moose-dev/meta.yaml                 |

### Versioner versions

Found 9 packages, 5 changed, 0 failed

| package     | status   | hash    | to hash   | version                       | to version                    |
|-------------|----------|---------|-----------|-------------------------------|-------------------------------|
| mpi         | CHANGE   | fcdc117 | cb02bf8   | 2025.06.13                    | 2025.07.23                    |
| libmesh-vtk | CHANGE   | 16265eb | 0858df0   | 9.4.2 build 3                 | 9.4.2 build 4                 |
| petsc       | CHANGE   | 9531950 | c47f7c3   | 3.23.0.6.gd9d7fd11dca build 2 | 3.23.0.6.gd9d7fd11dca build 3 |
| libmesh     | CHANGE   | 985bf51 | ac50710   | 2025.06.25 build 0            | 2025.06.25 build 1            |
| moose-dev   | CHANGE   | b5c6af5 | 92d6b4f   | 2025.06.26                    | 2025.07.22                    |

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/3066045/), step Docs: sync website on 9482074 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/31104/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/3066060/), step Generate coverage on 9482074 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>9241cb</th>
      <th colspan=4>#31104 948207</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.53%</td>
      <td>85.53%</td>
      <td>+0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>115835</td>
      <td>115836</td>
      <td>+1</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>19598</td>
      <td>19597</td>
      <td>-1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/31104/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/31104/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/31104/coverage/xfem)

</details>



This comment will be updated on new commits.

loganharbour committed:
add setopt rematchpcre to MPI Conda

Fixes zsh on linux with the conda packages

refs #31041

loganharbour committed:
Update versioner hashes

loganharbour commented:
> If this is merged, how long do the new containers take to be deployed to CIVET?

They'll be in tomorrow morning.

loganharbour commented:
Failure is unrelated - timeout.

loganharbour commented on conda/mpi/base_build.sh:
```diff
@@ -61,6 +62,14 @@ function baked_flags()
+    b_CPPFLAGS=$(echo "\$b_CPPFLAGS" | sed 's/-Wl,[[:space:]]//g')
+    b_CFLAGS=$(echo "\$b_CFLAGS" | sed 's/-Wl,[[:space:]]//g')
+    b_FFLAGS=$(echo "\$b_FFLAGS" | sed 's/-Wl,[[:space:]]//g')
+    b_LDFLAGS=$(echo "\$b_LDFLAGS" | sed 's/-Wl,[[:space:]]//g')
```
Do we still need this? Can add to https://github.com/idaholab/moose/pull/31935

hugary1995 replied:
Yes, please!

loganharbour replied:
Does this work in bash as expected?
